File: FrontEnd/entities/model/read-item/read-item.ts
import { ReadItemType } from "./read-item-type";
import { type Tag } from "../tag/tag";
import { Chapter } from "../chapter/chapter";

export interface ReadItem {
  id: string;
  title: string;
  description: string;
  type: ReadItemType;
  coverImage: string;
  tags: Array<Tag>;
  chapters: Array<Chapter>;
}

File: BackEnd/Catalog/DatabaseAccessLayer/Entities/ReadItem.cs
namespace DatabaseAccessLayer.Entities
{
    public class ReadItem : Entity
    {
        public string Title { get; set; }
        //enum on layer higher up, but here is string
        public string Type { get; set; }

        public string Description { get; set; }

        public byte[] CoverImage { get; set; }

        public float Rating { get; set; }

        public ICollection<ReadItemTag> ReadItemTags { get; set; }

        public ICollection<Chapter> Chapters { get; set; }
    }
}
